Lib/typing.py

```diff
@@ -230,16 +230,7 @@ def _remove_dups_flatten(parameters):
         else:
             params.append(p)
     # Weed out strict duplicates, preserving the first of each occurrence.
-    all_params = set(params)
-    if len(all_params) < len(params):
-        new_params = []
-        for t in params:
-            if t in all_params:
-                new_params.append(t)
-                all_params.remove(t)
-        params = new_params
-        assert not all_params, all_params
-    return tuple(params)
+    return tuple(dict.fromkeys(params))
```

Comment:
TBH this looks like a hack. If at some point we would make sets ordered the same way as dicts are, it would be good to update the code, but currently I think I like the old way more.